uni/lib/view/course_unit_info/widgets/course_unit_sheet.dart

```diff
@@ -1,106 +1,267 @@
+            child: Row(
+              children: [
+                FutureBuilder<File?>(
+                  builder:
+                      (BuildContext context, AsyncSnapshot<File?> snapshot) =>
+                          _buildAvatar(snapshot, 20),
+                  future: ProfileProvider.fetchOrGetCachedProfilePicture(
+                    session,
+                    studentNumber: int.parse(professor.value.code),
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.only(
+                    left: 10,
+                  ),
+                  child: Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      Text(
+                        professor.value.name,
+                        style: const TextStyle(fontSize: 14),
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+          );
+        }),
+      ],
+    ),
+  );
+}
+
+Widget buildBooksRow(BuildContext context, List<Book> books) {
+  return SizedBox(
+    height: 500,
+    width: double.infinity,
+    child: Wrap(
+      alignment: WrapAlignment.spaceBetween,
+      children: [
+        ...books.asMap().entries.map((book) {
+          return FutureBuilder<Image?>(
+            builder: (BuildContext context, AsyncSnapshot<Image?> snapshot) {
               return Padding(
-                padding: const EdgeInsets.only(bottom: 10),
-                child: Table(
-                  border: TableBorder.all(),
-                  children: rows
-                      .map(
-                        (e) => TableRow(
-                          children: e.children
-                              .sublist(0, min(4, e.children.length))
-                              .map(
-                                (e) => TableCell(
-                                  child: Padding(
-                                    padding: const EdgeInsets.all(8),
-                                    child: HtmlWidget(
-                                      e.outerHtml,
-                                    ),
-                                  ),
-                                ),
-                              )
-                              .toList(),
-                        ),
-                      )
-                      .toList(),
+                padding: const EdgeInsets.only(left: 15, right: 15, top: 10),
+                child: Column(
+                  children: [
+                    SizedBox(
+                      width: 135,
+                      height: 140, // adjust this value as needed
+                      child: snapshot.data ??
+                          const Image(
+                            image: NetworkImage(
+                              'https://nidcap.org/wp-content/uploads/2021/03/book.png',
```

Comment:
Not a good idea to depend on an online resource in this case. I suggest moving it to asset